File: app/src/main/java/com/tobiasfried/brewkeeper/DateRangeDialog.java
package com.tobiasfried.brewkeeper;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.appeaser.sublimepickerlibrary.SublimePicker;
import com.appeaser.sublimepickerlibrary.datepicker.SelectedDate;
import com.appeaser.sublimepickerlibrary.datepicker.SublimeDatePicker;
import com.appeaser.sublimepickerlibrary.helpers.SublimeListenerAdapter;
import com.appeaser.sublimepickerlibrary.helpers.SublimeOptions;
import com.appeaser.sublimepickerlibrary.recurrencepicker.SublimeRecurrencePicker;

import java.text.DateFormat;
import java.util.Locale;
import java.util.TimeZone;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

public class DateRangeDialog extends DialogFragment {

    // Date & Time formatter used for formatting
    // text on the switcher button
    DateFormat mDateFormatter, mTimeFormatter;

    // Picker
    SublimePicker mSublimePicker;

    // Callback to activity
    Callback mCallback;

    SublimeListenerAdapter mListener = new SublimeListenerAdapter() {
        @Override
        public void onCancelled() {
            if (mCallback!= null) {
                mCallback.onCancelled();
            }

            // Should actually be called by activity inside `Callback.onCancelled()`
            dismiss();
        }

        @Override
        public void onDateTimeRecurrenceSet(SublimePicker sublimeMaterialPicker,
                                            SelectedDate selectedDate,
                                            int hourOfDay, int minute,
                                            SublimeRecurrencePicker.RecurrenceOption recurrenceOption,
                                            String recurrenceRule) {
            if (mCallback != null) {
                mCallback.onDateTimeRecurrenceSet(selectedDate,
                        hourOfDay, minute, recurrenceOption, recurrenceRule);
            }

            // Should actually be called by activity inside `Callback.onCancelled()`
            dismiss();
        }
        // You can also override 'formatDate(Date)' & 'formatTime(Date)'
        // to supply custom formatters.
    };

    public DateRangeDialog() {
        // Initialize formatters
        mDateFormatter = DateFormat.getDateInstance(DateFormat.MEDIUM, Locale.getDefault());
        mTimeFormatter = DateFormat.getTimeInstance(DateFormat.SHORT, Locale.getDefault());
        mTimeFormatter.setTimeZone(TimeZone.getTimeZone("GMT+0"));
    }

    // Set activity callback
    public void setCallback(Callback callback) {
        mCallback = callback;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        mSublimePicker = (SublimePicker) getLayoutInflater().inflate(R.layout.dialog_date_range, container);

        // Retrieve SublimeOptions
        Bundle arguments = getArguments();
        SublimeOptions options = null;

        // Options can be null, in which case, default
        // options are used.
        if (arguments != null) {
            options = arguments.getParcelable("SUBLIME_OPTIONS");
        }

        mSublimePicker.initializePicker(options, mListener);
        return mSublimePicker;
    }

    // For communicating with the activity
    public interface Callback {
        void onCancelled();

        void onDateTimeRecurrenceSet(SelectedDate selectedDate,
                                     int hourOfDay, int minute,
                                     SublimeRecurrencePicker.RecurrenceOption recurrenceOption,
                                     String recurrenceRule);
    }

}

File: app/src/main/java/com/tobiasfried/rasa/DateRangeDialog.kt
package com.tobiasfried.rasa

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.appeaser.sublimepickerlibrary.SublimePicker
import com.appeaser.sublimepickerlibrary.datepicker.SelectedDate
import com.appeaser.sublimepickerlibrary.helpers.SublimeListenerAdapter
import com.appeaser.sublimepickerlibrary.helpers.SublimeOptions
import com.appeaser.sublimepickerlibrary.recurrencepicker.SublimeRecurrencePicker

import java.text.DateFormat
import java.util.Locale
import java.util.TimeZone
import androidx.fragment.app.DialogFragment

class DateRangeDialog : DialogFragment() {

    // Date & Time formatter used for formatting
    // text on the switcher button
    internal var mDateFormatter: DateFormat
    internal var mTimeFormatter: DateFormat

    // Picker
    internal var mSublimePicker: SublimePicker

    // Callback to activity
    internal var mCallback: Callback? = null

    internal var mListener: SublimeListenerAdapter = object : SublimeListenerAdapter() {
        override fun onCancelled() {
            if (mCallback != null) {
                mCallback!!.onCancelled()
            }

            // Should actually be called by activity inside `Callback.onCancelled()`
            dismiss()
        }

        override fun onDateTimeRecurrenceSet(sublimeMaterialPicker: SublimePicker,
                                             selectedDate: SelectedDate,
                                             hourOfDay: Int, minute: Int,
                                             recurrenceOption: SublimeRecurrencePicker.RecurrenceOption,
                                             recurrenceRule: String) {
            if (mCallback != null) {
                mCallback!!.onDateTimeRecurrenceSet(selectedDate,
                        hourOfDay, minute, recurrenceOption, recurrenceRule)
            }

            // Should actually be called by activity inside `Callback.onCancelled()`
            dismiss()
        }
        // You can also override 'formatDate(Date)' & 'formatTime(Date)'
        // to supply custom formatters.
    }

    init {
        // Initialize formatters
        mDateFormatter = DateFormat.getDateInstance(DateFormat.MEDIUM, Locale.getDefault())
        mTimeFormatter = DateFormat.getTimeInstance(DateFormat.SHORT, Locale.getDefault())
        mTimeFormatter.timeZone = TimeZone.getTimeZone("GMT+0")
    }

    // Set activity callback
    fun setCallback(callback: Callback) {
        mCallback = callback
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mSublimePicker = inflater.inflate(R.layout.dialog_date_range, container) as SublimePicker

        // Retrieve SublimeOptions
        val arguments = arguments
        var options: SublimeOptions? = null

        // Options can be null, in which case, default
        // options are used.
        if (arguments != null) {
            options = arguments.getParcelable("SUBLIME_OPTIONS")
        }

        mSublimePicker.initializePicker(options, mListener)
        return mSublimePicker
    }

    // For communicating with the activity
    interface Callback {
        fun onCancelled()

        fun onDateTimeRecurrenceSet(selectedDate: SelectedDate,
                                    hourOfDay: Int, minute: Int,
                                    recurrenceOption: SublimeRecurrencePicker.RecurrenceOption,
                                    recurrenceRule: String)
    }

}
